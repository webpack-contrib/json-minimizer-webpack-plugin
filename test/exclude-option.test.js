const JsonMinimizerPlugin = require("../src/index");

const {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} = require("./helpers");

describe("exclude option", () => {
  let compiler;

  beforeEach(() => {
    const testJsonId = "./include-exclude/*.json";

    compiler = getCompiler(testJsonId);
  });

  it("should match snapshot for a single RegExp value exclude", async () => {
    new JsonMinimizerPlugin({
      exclude: /include-exclude\/exclude/i,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple RegExp values exclude", async () => {
    new JsonMinimizerPlugin({
      exclude: [
        /include-exclude\/exclude-(1)/i,
        /include-exclude\/exclude-(2)/i,
      ],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple String values exclude", async () => {
    new JsonMinimizerPlugin({
      exclude: ["include-exclude/exclude-1", "include-exclude/exclude-2"],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
