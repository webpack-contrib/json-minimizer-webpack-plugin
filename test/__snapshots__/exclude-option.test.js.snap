// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exclude option should match snapshot for a single RegExp value exclude: assets 1`] = `
{
  "include-exclude/exclude-1.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/exclude-2.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/include-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-3.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-4.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
}
`;

exports[`exclude option should match snapshot for a single RegExp value exclude: errors 1`] = `[]`;

exports[`exclude option should match snapshot for a single RegExp value exclude: warnings 1`] = `[]`;

exports[`exclude option should match snapshot for multiple RegExp values exclude: assets 1`] = `
{
  "include-exclude/exclude-1.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/exclude-2.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/include-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-3.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-4.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
}
`;

exports[`exclude option should match snapshot for multiple RegExp values exclude: errors 1`] = `[]`;

exports[`exclude option should match snapshot for multiple RegExp values exclude: warnings 1`] = `[]`;

exports[`exclude option should match snapshot for multiple String values exclude: assets 1`] = `
{
  "include-exclude/exclude-1.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/exclude-2.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true
}
",
  "include-exclude/include-1.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-2.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-3.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
  "include-exclude/include-4.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
}
`;

exports[`exclude option should match snapshot for multiple String values exclude: errors 1`] = `[]`;

exports[`exclude option should match snapshot for multiple String values exclude: warnings 1`] = `[]`;
